pandas/core/frame.py

```diff
@@ -8988,6 +8988,42 @@ def aggregate(self, func=None, axis: Axis = 0, *args, **kwargs):
 
     agg = aggregate
 
+    @overload
+    def any(
+        self: DataFrame,
+        *,
+        axis: Axis = ...,
+        bool_only: bool | None = ...,
+        skipna: bool = ...,
+        level: None = ...,
+        **kwargs,
+    ) -> Series:
+        ...
+
+    @overload
+    def any(
+        self: DataFrame,
+        *,
+        axis: Axis = ...,
+        bool_only: bool | None = ...,
+        skipna: bool = ...,
+        level: Level,
+        **kwargs,
+    ) -> DataFrame | Series:
+        ...
+
+    @doc(NDFrame.any, **_shared_doc_kwargs)
+    def any(
```

Comment:
# error: Signature of "any" incompatible with supertype "NDFrame"  [override]

and same for pandas/core/series.py:4400: error: Signature of "any" incompatible with supertype "NDFrame"  [override]

Reply:
Please check the output of mypy and add mypy ignore accordingly.

1. You can remove all of the self annotations (no need for `self: DataFrame` and `self: Series`)
2. We cannot yet have `*` for the implementation, only in the overload definitions
3. At the line of the first overload add the following ignore (all other lines do not need ignores):
```py
    # error: Signature of "any" incompatible with supertype "NDFrame"  [override]
    @overload  # type: ignore[override]
    def any(
```
4. run mypy locally (or let the CI run mypy)